### PR TITLE
Replace Koalas with pandas API on Spark

### LINKED ISSUE #1864: Update koalas code to pyspark pandas API instead in Featuretools
The koalas library is now in maintenance mode only, and the core code has been integrated into pyspark, starting with version 3.2. We should update our koalas specific code to use the new pyspark pandas API: https://spark.apache.org/docs/latest/api/python/getting_started/quickstart_ps.html

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #1864

### CODE REVIEW

File: featuretools/utils/spark_utils.py
Comment (jeff-hernandez) [inline featuretools/utils/spark_utils.py:15]: Spark series  doesn't support an array that contains floats and integers

```python
ps.from_pandas(pd.Series([[0.0, 0.0], [7.0, 3.0], [14.0, 6.0], [-1, -1], [-1, -1]]))
```

```
TypeError: element in array field 0: DoubleType can not accept object -1 in type <class 'int'>
```

File: featuretools/entityset/entityset.py
Comment (jeff-hernandez) [inline featuretools/entityset/entityset.py:1584]: Spark doesn't support range type for column assignment

File: setup.cfg
Comment (jeff-hernandez) [inline setup.cfg:49]: We'll need to change this to the corresponding woodwork version before merging

**Comment by jeff-hernandez** [comment]
we'll need to switch the required unit tests from koalas to spark

File: docs/source/release_notes.rst
Comment (rwedge) [inline docs/source/release_notes.rst:231]: I don't think we need to retroactively replace Koalas with Spark in the changelog

Comment (jeff-hernandez) [reply docs/source/release_notes.rst:231]: yep agreed, I missed that change possibly after merging conflicts locally

File: setup.cfg
Comment (rwedge) [inline setup.cfg:92]: are we able to remove the pandas<1.4 constraint now?

File: spark-requirements.txt
Comment (rwedge) [inline spark-requirements.txt:1]: is this file still necessary?

File: setup.cfg
Comment (jeff-hernandez) [reply setup.cfg:92]: not yet  there are still breaking changes

File: spark-requirements.txt
Comment (jeff-hernandez) [reply spark-requirements.txt]: nope removed

File: setup.cfg
Comment (rwedge) [reply setup.cfg:89]: are we tracking that issue somewhere?

Comment (jeff-hernandez) [reply setup.cfg:89]: it seems like it, opened #1951

File: featuretools/tests/entityset_tests/test_last_time_index.py
Comment (thehomebrewnerd) [inline featuretools/tests/entityset_tests/test_last_time_index.py:183]: Can we update this to use the `if es.dataframe_type == Library.SPARK.value` approach that we are using elsewhere in this file? There are few locations in this file that could be updated.

File: .github/workflows/unit_tests_with_latest_deps.yml
Comment (jeff-hernandez) [reply .github/workflows/unit_tests_with_latest_deps.yml]: updated

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Comment by jeff-hernandez** [comment]
I think we'll need to merge and release the changes in Woodwork first before doing the same in featuretools.

**Comment by thehomebrewnerd** [comment]
@jeff-hernandez Just a quick heads-up. The `compatibility` attribute for the primitives added by #1948 will need to be updated as well when you fix the release notes merge conflict.

File: setup.cfg
Comment (thehomebrewnerd) [inline setup.cfg:88]: Do we need an upper bound here?

Comment (jeff-hernandez) [reply setup.cfg]: not necessary, removed

**Review by thehomebrewnerd — APPROVED** [review]
LGTM, assuming tests pass.

**Pull request merged**
